updated for 0.6

ExpandingMan opened the pull request:
I found myself needing this package so I updated it for 0.6 and switched to the most up-to-date yaml I could find, found [here](https://github.com/ua-parser/uap-core/blob/master/regexes.yaml).  Some of the tests had to be adjusted for the new yaml.

In my application performance is a concern, so ideally I'd like to rewrite this to be more type-stable, but it won't be a priority for me if what I have here is adequate.

randyzwitch commented on README.md:
```diff
@@ -10,9 +10,14 @@ UAParser is a Julia port of [ua-parser](https://github.com/tobie/ua-parser), whi
 
-UAParser is a limited Julia implementation heavily influenced by the [Python code](https://github.com/tobie/ua-parser/tree/master/py) from the ua-parser library.
+UAParser is a limited Julia implementation heavily influenced by the [Python code](https://github.com/tobie/ua-parser/tree/master/py) from the ua-parser
+library.  Ideally the code would be rewritten to be more performant and Julian, but right now this is not a priority.  This should already be much more
```
Not sure it's necessary to comment on the perceived flaws of the Python library, nor that an "ideal" is that the code be re-written.

randyzwitch commented on README.md:
```diff
@@ -59,6 +64,6 @@ You can index into the results of these functions like any other Julia composite
+## Licensing
 
-The data contained in regexes.yaml is Copyright 2009 Google Inc. and available under the Apache License, Version 2.0. The licensing of the UAParser Julia module is under the [default MIT Expat license](https://github.com/JuliaWeb/UAParser.jl/blob/master/LICENSE.md).
+The licensing of the UAParser Julia module is under the [default MIT Expat license](https://github.com/JuliaWeb/UAParser.jl/blob/master/LICENSE.md).
```
Please retain the Google and Apache notices. While they may be redundant, IMO it's better to overcredit than to assert that these files can be relicensed.

randyzwitch commented on src/UAParser.jl:
```diff
@@ -276,75 +358,80 @@ function parseos(user_agent_string::AbstractString)
+
+    temp["device"] = String[ξ.family for ξ ∈ x]
+
+    temp["brand"] = String[nothing_to_emptystr(ξ.brand) for ξ ∈ x]
```
This use of unicode seems excessive, given the simplicity of the list comprehension. My preference is to put this back to ASCII, so that the next developer who comes along and may be less mathematically inclined will not have to understand the set syntax you are using.

ExpandingMan commented:
Ok, I've done a number of things.  I made the changes you suggested.  I also switched everything over to use [Nulls.jl](https://github.com/JuliaData/Nulls.jl) instead of `nothing`.  This allows this package to play much nicer with the new data ecosystem.  I've also cleaned up the tests quite a bit.

Let me know if you think this is ok, things work very similarly with Nulls, but I know the use of Nulls is a relatively new thing.  I'm sure once the new releases of DataFrames and related are tagged, it'll be clear that this is preferable.

randyzwitch commented on src/UAParser.jl:
```diff
@@ -189,162 +268,153 @@ function parseuseragent(user_agent_string::AbstractString)
+function DataFrame(x::AbstractVector{DeviceResult})
+    temp = DataFrame()
+
+    temp["device"] = String[element.family for element ∈ x]
```
Please use `in` for the list comprehensions

ExpandingMan commented:
Ok, I have made the requested changes.  There seemed to be no appveyor yaml so I added one.  Tests are now on 0.6 and nightly.

You may want to wait until DataFrames gets tagged again to merge this; I'm hoping that will happen very soon.

randyzwitch commented:
@ExpandingMan LGTM, thanks! Can you ping this thread when the dataframes changes get tagged? I don't follow that closely. Or, if you know which tag its going to be, you could set REQUIRES to that version

ExpandingMan commented:
Sure, I'll try to remember to do that.

randyzwitch commented:
@ExpandingMan I think for this, you should update REQUIRES for dataframes 0.11, then this will be good for merging and a new version for METADATA

randyzwitch commented:
Nulls.jl was also renamed to Missings.jl, so that will need to be fixed before this can be merged to master and tagged

ExpandingMan commented:
Sorry, I forgot about this after the first email.  Was this ok to merge?  Should I change it, it's still using Nulls?

randyzwitch commented:
I just switched it, but then merged to master by accident. I'll fix any outstanding issues.

ExpandingMan commented:
Ok, should be able to just do a global search and replace `Null -> Missing`, `null -> missing`, but be a little careful with that latter, sometimes there are `null`s in names somewhere not related to Nulls.jl.

If you want me to make a commit I can do it right now.

randyzwitch commented:
If you could do it right now, that'd be great. I mistakenly thought I could just switch REQUIRE, but I didn't see that the code itself changed from `Null` to `Missing`.

ExpandingMan commented:
Done!

randyzwitch commented:
I think you need a new PR, since this one got merged.
